caikit/runtime/service_generation/protoable.py

```diff
@@ -16,15 +16,26 @@
 """
 
 # Standard
-from typing import Dict, Type, Union, get_args, get_origin
+from typing import Dict, List, Type, Union, get_args, get_origin
 import typing
 
 # First Party
+from py_to_proto.dataclass_to_proto import Annotated, OneofField
 import alog
 
 # Local
-from caikit.core.data_model.dataobject import DATAOBJECT_PY_TO_PROTO_TYPES
+from caikit.core.data_model.base import DataBase
+from caikit.core.data_model.dataobject import (
+    DATAOBJECT_PY_TO_PROTO_TYPES,
+    make_dataobject,
+)
 from caikit.runtime.service_generation.type_helpers import is_data_model_type
+import caikit
+
+# This global holds the mapping of union list types to their respective
+# UnionList wrappers so that the same message is not recreated
+# unnecessarily
+_UNION_LIST_SOURCE_TYPES = {}
```

Comment:
Does this need to be pre-populated with the static ones?

Reply:
Since we're not creating the static ones anymore, this doesn't populate the static ones. This only has the custom ones that get created through unions